kaolin/datasets/modelnet.py

```diff
@@ -47,143 +41,129 @@ class ModelNet10(torch.utils.data.Dataset):
-        download (bool): downloads the dataset if not found in root
+        download (bool): downloads the dataset if not found in basedir
         categories (str): list of object classes to be loaded
-        single_view (bool): if true only one roation is used, if not all 12 views are loaded
 
     Returns:
         .. code-block::
-        
+
         dict: {
             'attributes': {'name': str, 'class': str},
             'data': {'voxels': torch.Tensor}
         }
 
 
     Examples:
-        >>> dataset = ModelNet(root='../data/')
+        >>> dataset = ModelNet(basedir='../data/')
         >>> train_loader = DataLoader(dataset, batch_size=10, shuffle=True, num_workers=8)
         >>> obj = next(iter(train_loader))
         >>> obj['data']['data']
         torch.Size([10, 30, 30, 30])
     """
 
-    def __init__(self, root: str = '../data/', train: bool = True, test: bool = True,
-                 download: bool = True, categories: list = ['chair'], single_view: bool = True):
-        if not os.path.exists(root + '/ModelNet/'):
-            assert download, "ModelNet is not found, and download is set to False"
-            assert (train or test), 'either train or test must be set to True'
-
-        if not single_view:
-            side = ''
-        else:
-            side = '_1'
-
-        all_classes = [os.path.basename(os.path.dirname(c))
-                       for c in glob(root + '/ModelNet/volumetric_data/*/')]
-        self.names = []
-        for category in categories:
-            assert category in all_classes, 'object class {0} not in list of availible classes: {1}'.format(
-                category, all_classes)
-            if train:
-                self.names += glob(
-                    root + '/ModelNet/volumetric_data/{0}/*/{1}/*{2}.mat'.format(category, 'train', side))
-            if test:
-                self.names += glob(
-                    root + '/ModelNet/volumetric_data/{0}/*/{1}/*{2}.mat'.format(category, 'test', side))
+    def __init__(self, basedir: str = 'data/ModelNet/', cache_dir: str = 'cache', train: bool = True, categories: list = ['bed'],
+                 resolutions: List[int] = [32], device: str = 'cpu'):
+
+        self.basedir = basedir
+        self.device = torch.device(device)
+        self.cache_dir = cache_dir
+        self.params = {'resolutions': resolutions}
+        self.cache_transforms = {}
+
+        mesh_dataset = ModelNet(basedir, train, categories, device)
+
+        self.names = mesh_dataset.names
+        self.categories = mesh_dataset.categories
+        self.cat_idxs = mesh_dataset.cat_idxs
+
+        for res in self.params['resolutions']:
+            self.cache_transforms[res] = tfs.CacheCompose([
+                tfs.TriangleMeshToVoxelGrid(res, normalize=True, vertex_offset=0.5),
+                tfs.FillVoxelGrid(thresh=0.5),
+                tfs.ExtractProjectOdmsFromVoxelGrid()
+            ], self.cache_dir)
```

Comment:
so `cache_dir` by default is `cache` which is relative to the current directory, which can lead to bad surprise. I suggest to put `cache_dir = None` by default and if `cache_dir is None` then `self.cache_dir = os.path.join(self.basedir + 'cache')`.

Reply:
Great suggestion, implemented.